pelican/tests/test_pelican.py

```diff
@@ -35,6 +35,22 @@ def recursiveDiff(dcmp):
     return diff
 
 
+def assertDirsEqual(left_path, right_path):
+    """
+    Raise AssertionError if files in paths differ (ignoring whitespace).
+    """
+    proc = subprocess.Popen(
+        ['git', '--no-pager', 'diff', '--no-ext-diff', '--exit-code',
+         '-w', left_path, right_path],
+        stdout=subprocess.PIPE,
+        stderr=subprocess.STDOUT
+    )
+
+    out = proc.communicate()
+    if proc.returncode:
+        assert not out, out
```

Comment:
I thought it strange too; however, it's consistent with what was there before, and I'm attempting to change as little as possible.  

That said, I don't see any problem with this suggestion and it's more clear and terse; I can include this change if you would prefer.

Reply:
Yeah, I would prefer changing it. This way it is a bit duplicated. `if proc.returncode:` already implies that out will be non empty and `assert` will fail. So the actual test is indeed `proc.returncode == 0`.

BTW, `proc.communicate()` returns a tuple `(stdout, stderr)`. Even if they are both empty, `('', '')` will always be True. So, `assert not out` is identical to `assert False` (i.e. always fail) :).

Reply:
Oh right, thanks 😅.  For some reason I think I was thinking that because I merged the out and err streams in the subprocess call, I would only get one stream back (rather than ``(bytes, None)``).